devito/data/data.py

```diff
@@ -210,13 +316,56 @@ def _normalize_index(self, idx):
             if any(i is Ellipsis for i in idx):
                 # Explicitly replace the Ellipsis
                 items = (slice(None),)*(self.ndim - len(idx) + 1)
-                return idx[:idx.index(Ellipsis)] + items + idx[idx.index(Ellipsis)+1:]
+                items = idx[:idx.index(Ellipsis)] + items + idx[idx.index(Ellipsis)+1:]
+            else:
+                items = idx + (slice(None),)*(self.ndim - len(idx))
+            # Normalize slice steps:
+            processed = [slice(i.start, i.stop, 1) if
+                         (isinstance(i, slice) and i.step is None)
+                         else i for i in items]
+            return as_tuple(processed)
+
+    def _process_args(self, idx, val):
+        """If comm_type is parallel we need to first retrieve local unflipped data."""
+        if any(isinstance(i, slice) and i.step is not None and i.step < 0
+               for i in as_tuple(idx)):
+            processed = []
+            transform = []
+            for j, k in zip(idx, self._distributor.glb_shape):
+                if isinstance(j, slice) and j.step is not None and j.step < 0:
+                    if j.start is None:
+                        stop = None
+                    else:
+                        stop = j.start + 1
+                    if j.stop is None and j.start is None:
+                        start = int(np.mod(k-1, -j.step))
+                    elif j.stop is None:
+                        start = int(np.mod(j.start, -j.step))
+                    else:
+                        start = j.stop + 1
+                    processed.append(slice(start, stop, -j.step))
+                    transform.append(slice(None, None, np.sign(j.step)))
+                else:
+                    processed.append(j)
+            if isinstance(val, Data) and len(transform) > 0 and \
+                    len(val._distributor.shape) > len(val.shape):
+                distributor = \
```

Comment:
nitpicking: don't need the \ ? you can inline the rhs here and split each argument over multiple lines

Reply:
I think it is needed no?